CONTRIBUTING.md

```diff
@@ -26,26 +26,156 @@
+Working on a task means moving it from Approved to Doing, this operation is only allowed if it doesn't violate the Doing WIP limit. All available work is found on the Waffle board in Approved, the next task is always at the top. Assign yourself as the owner of the issue at the top and move it from the top of Approved to the bottom of Doing.
+
+If a task includes an existing pull request from someone outside the engineering team, they should not be assigned as the owner. All work must be owned by the engineering team, if someone on another team e.g. support has provided code then work with them to get it merged but ultimately it is the responsibility of the engineering team to manage the flow of work and ensure all aspects of a card are completed.
+
+## Doing
+#### Fixing an issue
+The issue description contains a list of the steps required to close an issue, this is added by the [issue template](.github/ISSUE_TEMPLATE.md). As each step is completed, update the description.
+
+##### New issues discovered while working an issue
+If a new bug or feature requirement is found while working on an assigned issue, that new work item should be logged as a separate issue in Backlog.
+
+If the new issue blocks the existing issue, you should assign yourself to the new issue and move it to Doing. The existing issue should be updated with all the work that has been done, the label 'status-blocked' added and a link to the blocking issue should be included in the issue description. In this situation the WIP limit can be exceeded. Similar to situations where the WIP limit prevents new work being started the team needs to work to reduce the WIP so that new tasks can be started.
+
+If the new issue does not prevent the resolution of your current task it will be treated as any new issue.
+
+##### Creating a pull request
+If an issue affects multiple branches the owner should always start by creating a pull request for the newest branch, generally this means master. Only one pull request should ever be in progress for a task in Doing.
+
+To ensure that issues and pull request are correctly associated, reference the issue at the end of the commit message, e.g.:
+    Connected https://github.com/rcbops/rpc-openstack/issues/0
+
+## Doing -> Needs review
+A card can only be moved from Doing to Needs-review if Needs-review is below its WIP limit. If that condition is met and you are ready for feedback on your pull request, move the card (this should be a connected pair of the issue and pull request) to the bottom of Needs-review.
+
+## Needs review
+Reviewers should always review the tasks in the order the are listed in Needs-review. If owners need to make changes to the pull request, for example base on feedback from reviewers, the task should remain in Needs-review while that is done. The reviewer merging the commit should delete the issue branch (using the link on the pull request page) and tick the issue checklist for the fixed branch.
```

Comment:
How are you defining oldest to newest? I haven't seen any way to sort columns based on some property of the cards.

The idea behind always adding new cards to the bottom of Needs-review is so that the card at the top of the column is always the oldest (the card that has been in that column the longest). This reduces the chances that cards, that are difficult to review, will sit there for a long time because the will block other reviews.